components/access-management/kfam/bindings.go

```diff
@@ -57,7 +57,7 @@ type BindingClient struct {
 //getBindingName returns bindingName, which is combination of user kind, username, RoleRef kind, RoleRef name.
 func getBindingName(binding *Binding) (string, error) {
 	// Only keep lower case letters and numbers, replace other with -
-	reg, err := regexp.Compile("[^a-z0-9]+")
+	reg, err := regexp.Compile("[^a-zA-Z0-9]+")
```

Comment:
I'm not aware of all configurations, but in ours Azure AD we're getting uppercased username by default in User Principal Name (UPN) claim. So if we will try to add contributor to some namespace, then the name of the binding for the user `TEST@example.com` would be `user-example-com-clusterrole-edit` instead of `user-test-example-com-clusterrole-edit`

Reply:
I would suggest we leave this change out of the PR and scope it down to just porting what we already have to AuthorizationPolicy. If you need this new feature, let's discuss on a new issue.

Reply:
It looks very minor change for new issue, but ok, you're the boss here, I'll create new one.